semgrep/tests/performance/test_semgrep_rules_repo_perf.py

```diff
@@ -221,20 +221,27 @@
 
 
 @pytest.fixture(scope="session")
-def semgrep_rules_repo(tmp_path_factory):
+def semgrep_rules_repo(request, tmp_path_factory):
     repo_path = tmp_path_factory.mktemp("repo")
+
+    getoption = request.config.getoption
+    is_disabled = getoption("benchmark_skip") or getoption("benchmark_disable")
+    is_force_enabled = getoption("benchmark_enable") or getoption("benchmark_only")
+    if is_disabled and not is_force_enabled:
+        return repo_path  # not cloning since we're not gonna run benchmarks
+
     subprocess.check_output(
         ["git", "clone", "https://github.com/returntocorp/semgrep-rules", repo_path]
     )
     subprocess.check_output(
-        ["git", "--git-dir", repo_path / ".git", "checkout", "c3196b4"]  # May 16, 2020
+        ["git", "checkout", "c3196b4"], cwd=repo_path  # May 16, 2020
     )
 
-    yield repo_path
+    return repo_path
 
 
 @pytest.fixture(params=RULE_PATHS.strip().splitlines())
-def semgrep_rules_rule(semgrep_rules_repo, request, tmp_path):
+def semgrep_rules_rule(semgrep_rules_repo, request, tmp_path, benchmark):
```

Comment:
Does this avoid running the fixture unless we're benchmarking? The fact that this variable is unused makes me think there's a better way to solve this.

Reply:
Yep, this makes pytest-benchmark decide if it should run. I'd need to research alternative ways to accomplish the same.